### DISCUSSION TITLE
Generate static payloads at build time using `useStaticData`

### DISCUSSION THREAD

**Discussion by maximilliangeorge** [nuxt/nuxt#25206] [Ideas]
This proposal is a follow up on an interesting discussion I had [on Discord](https://discord.com/channels/473401852243869706/1138217695171592192/1138580107439636491). I've been thinking about taking a stab at building this as a module, however I wanted to get some input on this before diving in, perhaps also checking to see if there are similar ideas already in the works.

The idea is basically to have add another hook for fetching data, similar to `useAsyncData`. However, this data would only be fetched at build time and the data would be served as a static JSON payload.

This allows mixing dynamic and static data with even further granularity. It also ensures the data is available right after build, with no need for an initial request for caching the payload.

To give a concrete example where I would use this: I'm building a page where navigation should happen instantly by means of rendering the header and page title first, while the page body is loaded async. Since the page header and title isn't expected to change very often I can safely use useStaticData for the headers, combined with scheduled builds.

An alternative approach to this could be to enable some sort of advanced cache rules for useAsyncData that lets you cache the request at build time rather than runtime. This might actually be preferable since I could then revalidate the payload in the background – however, I'm guessing it's a bit more complex to implement, especially as a Nuxt module.

**Comment by AlexVFornazieri** [comment]
Hey guys, any progress related to it?

I'm building a multiple sites CMS platform and want the public site was served with statics and only refetch when admin/content manager update something, trigger a build process

**Reply by maximilliangeorge** [reply]
Hey @AlexVFornazieri

I haven't had time to look into this further but I stil think it's an interesting idea. For your use case I think you can use the ordinary full-static mode, or perhaps routeRules if you need further granularity based on the route pattern.

Please upvote the OP if you want this feature, might help shine some light on it :)
